pkg/operator/starter.go

```diff
@@ -499,8 +461,83 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 		go backupController.Run(ctx, 1)
 	}
 
-	go masterMachineInformer.Run(ctx.Done())
+	// we have to wait for the definitive result of the cluster version informer to make the correct machine API decision
+	klog.Infof("waiting for cluster version informer sync...")
+	versionTimeoutCtx, versionTimeoutCancelFunc := context.WithTimeout(context.Background(), 5*time.Minute)
+	clusterVersionInformerSynced := cache.WaitForCacheSync(versionTimeoutCtx.Done(), clusterVersions.Informer().HasSynced)
+	versionTimeoutCancelFunc()
+
+	if !clusterVersionInformerSynced {
+		return fmt.Errorf("could not sync ClusterVersion, aborting operator start")
+	}
+
+	clusterMemberControllerInformers := []factory.Informer{masterNodeInformer}
+	machineLister := machinelistersv1beta1.NewMachineLister(masterMachineInformer.GetIndexer())
```

Comment:
On a cluster where the API isn't installed, having an informer for the Machines will cause errors won't it?